node/node.go

```diff
@@ -570,6 +598,51 @@ func (node *Node) RemovePeersHandler() {
 	}
 }
 
+//UpdateStakingList updates the stakes of every node.
+func (node *Node) UpdateStakingList(block *types.Block) error {
+	signerType := types.HomesteadSigner{}
+	txns := block.Transactions()
+	for i := range txns {
+		txn := txns[i]
+		value := txn.Value().Int64()
+		currentSender, _ := types.Sender(signerType, txn)
+		_, isPresent := node.CurrentStakes[currentSender]
+		toAddress := txn.To()
+		if *toAddress != node.StakingContractAddress { //Not a address aimed at the staking contract.
+			continue
+		}
+		if value > int64(0) { //If value >0 means its a staking deposit transaction
```

Comment:
let's make a switch case 

need some logic to exact the method call

Reply:
Added a issue: https://github.com/harmony-one/harmony/issues/430